akka-http-core/src/main/scala/akka/http/impl/engine/http2/RequestParsing.scala

```diff
@@ -117,12 +117,16 @@ private[http2] object RequestParsing {
             malformedRequest("Pseudo-header ':status' is for responses only; it cannot appear in a request")
 
           case ("content-type", ct) =>
-            val contentType = ContentType.parse(ct).right.getOrElse(malformedRequest(s"Invalid content-type: '$ct'"))
-            rec(remainingHeaders.tail, method, scheme, authority, pathAndRawQuery, contentType, contentLength, cookies, true, headers)
+            if (contentType eq ContentTypes.`application/octet-stream`) {
```

Comment:
But what if the header *does* actually contain `application/octet-stream`? Maybe we should use `OptionVal` for the `contentType` parameter instead to figure that out?